server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -80,14 +83,45 @@ def redeploy
     def validate
       @deployment.valid?
       render json: {
-          :validation => {
-              :deployment_id => @deployment.id,
-              :errors => @deployment.errors.full_messages,
-              :warnings => @deployment.warnings
-          }
+        :validation => {
+          :deployment_id => @deployment.id,
+          :errors => @deployment.errors.full_messages,
+          :warnings => @deployment.warnings
+        }
       }
     end
 
+    def validate_cdn
+      begin
+        if params.key?("cdn_url")
+          def ad_hoc_req(uri_str)
```

Comment:
Rubocop and I have some stylistic differences, but we worked it out. Also, he's pretty intimidating.